File: Backend/app/schemas/catalogs.py
from pydantic import BaseModel, ConfigDict


# ─── Year ───────────────────────────────────────────────────

class YearResponse(BaseModel):
    id:     int
    year:   int
    active: bool

    model_config = ConfigDict(from_attributes=True)


# ─── Month ──────────────────────────────────────────────────

class MonthResponse(BaseModel):
    id:              int
    month_name:      str
    month_initials:  str

    model_config = ConfigDict(from_attributes=True)


# ─── Status ─────────────────────────────────────────────────

class StatusResponse(BaseModel):
    id:     int
    name:   str
    active: bool

    model_config = ConfigDict(from_attributes=True)


# ─── Param ──────────────────────────────────────────────────

class ParamResponse(BaseModel):
    id:          int
    name:        str
    value:       float
    description: str
    active:      bool

    model_config = ConfigDict(from_attributes=True)


class ParamUpdateRequest(BaseModel):
    """Solo el valor es actualizable desde la interfaz."""
    value: float

File: Frontend/models/catalogs.ts
export interface Year {
  id: number
  year: number
  active: boolean
}

export interface Month {
  id: number
  month_name: string
  month_initials: string
}

export interface Status {
  id: number
  name: string
  active: boolean
}

export interface Param {
  id: number
  name: string
  value: number
  description: string
  active: boolean
}

export interface BaseCatalogsResponse {
  years: Year[]
  params: Param[]
  months?: Month[]
  status?: Status[]
}
